src/firestore/api.ts

```diff
@@ -329,6 +327,111 @@ export class FirestoreApi {
+    if (backups.length === 0) {
+      logger.info("No backups found.");
+      return;
+    }
+    const sortedBackups: Backup[] = backups.sort(sort.compareApiBackup);
+    const Table = require("cli-table");
+    const table = new Table({
+      head: ["Backup Name", "Database Name", "Snapshot Time", "State"],
+      colWidths: [
+        Math.max(...sortedBackups.map((backup) => backup.name!.length + 5), 20),
+        Math.max(...sortedBackups.map((backup) => backup.database!.length + 5), 20),
+        30,
+        10,
+      ],
+    });
+
+    table.push(
+      ...sortedBackups.map((backup) => [
+        this.prettyBackupString(backup),
+        this.prettyDatabaseString(backup.database || ""),
+        backup.snapshotTime,
+        backup.state,
+      ]),
+    );
+    logger.info(table.toString());
+  }
+
+  /**
+   * Print an array of backup schedules to the console as an ASCII table.
+   * @param backupSchedules the array of Firestore backup schedules.
+   * @param databaseId the database these schedules are associated with.
+   */
+  prettyPrintBackupSchedules(backupSchedules: BackupSchedule[], databaseId: string): void {
+    if (backupSchedules.length === 0) {
+      logger.info(`No backup schedules for database ${databaseId} found.`);
+      return;
+    }
+    const sortedBackupSchedules: BackupSchedule[] = backupSchedules.sort(
+      sort.compareApiBackupSchedule,
+    );
+    sortedBackupSchedules.forEach((schedule) => this.prettyPrintBackupSchedule(schedule));
+  }
+
+  /**
+   * Print important fields of a backup schedule to the console as an ASCII table.
+   * @param backupSchedule the Firestore backup schedule.
+   */
+  prettyPrintBackupSchedule(backupSchedule: BackupSchedule): void {
+    const Table = require("cli-table");
+    const table = new Table({
+      head: ["Field", "Value"],
+      colWidths: [25, Math.max(50, 5 + backupSchedule.name!.length)],
+    });
+
+    table.push(
+      ["Name", clc.yellow(backupSchedule.name!)],
+      ["Create Time", clc.yellow(backupSchedule.createTime!)],
+      ["Last Update Time", clc.yellow(backupSchedule.updateTime!)],
+      ["Retention", clc.yellow(backupSchedule.retention)],
+      ["Recurrence", this.prettyRecurrenceString(backupSchedule)],
+    );
+    logger.info(table.toString());
+  }
+
+  /**
+   * Returns a pretty representation of the Recurrence of the given backup schedule.
+   * @param {BackupSchedule} backupSchedule the backup schedule.
+   */
+  prettyRecurrenceString(backupSchedule: BackupSchedule): string {
+    if (backupSchedule.dailyRecurrence) {
+      return clc.yellow("DAILY");
+    } else if (backupSchedule.weeklyRecurrence) {
+      return clc.yellow(`WEEKLY (${backupSchedule.weeklyRecurrence.day})`);
+    }
+    return "";
+  }
+
+  /**
+   * Print important fields of a backup to the console as an ASCII table.
+   * @param backup the Firestore backup.
+   */
+  prettyPrintBackup(backup: Backup) {
+    const Table = require("cli-table");
```

Comment:
This runtime require is a bit strange - any particular reason for doing so here? If not, probably best to import this normally.

Reply:
The existing ones did too - I'll move them up to the top.